astropy/io/fits/connect.py

```diff
@@ -162,7 +163,10 @@ def read_table_fits(input, hdu=None):
 
     # TODO: deal properly with unsigned integers
 
-    for key, value, comment in table.header.cards:
+    # Read time coordinates as Time
+    hdr = FITS_time().read_time(table.header, t)
```

Comment:
It's not uncommon to change the way things are implemented once a PR evolves - specifically for a thing as complex as the FITS time standard. In this case, it's very useful to have a fresh pair of eyes looking at it.